python/cudf/cudf/core/dataframe.py

```diff
@@ -793,11 +793,30 @@ def fallback(col, fn):
                 else:
                     result[col] = fallback(rhs._cols[col], _reverse_op(fn))
         elif isinstance(other, Series):
-            raise NotImplementedError(
-                "Series to DataFrame arithmetic not supported "
-                "until strings can be used as indices. Try converting your"
-                " Series into a DataFrame first."
-            )
+            other_cols = other.to_pandas().to_dict()
+            result_cols = list(self.columns)
+            for new_col in other_cols.keys():
+                if new_col not in result_cols:
+                    result_cols.append(new_col)
+            for col in result_cols:
+                if col in self.columns and col in other.index:
+                    l_opr = self._cols[col]
+                    r_opr = other_cols[col]
+                else:
+                    if col not in self.columns:
+                        r_opr = other_cols[col]
```

Comment:
A much more efficient approach is probably to use `other.loc[col]` instead of converting it to a dict. I'm vaguely interested in benchmarking this in large cases before we ship. How do you others feel about that? Not worth the trouble? This implementation looks to be slower than Pandas.

Reply:
I believe that would actually be much slower than doing a dictionary lookup, but I could be mistaken.